torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2863,6 +3039,49 @@ bool canOmitStopPredicate(
   return true;
 }
 
+std::pair<Val*, Val*> hoistPredicates(
+    Val* start_index,
+    Val* stop_index,
+    const std::vector<kir::ForLoop*>& loops,
+    kir::ForLoop* unswitch_or_vec_loop,
+    IterDomain* predicated_consumer_id,
+    TensorView* predicated_consumer_tv,
+    TensorDomain* ref_td,
+    const std::unordered_map<IterDomain*, Val*>& ref_index_map) {
+  const std::pair<Val*, Val*> same_indices{start_index, stop_index};
+
+  // Don't hoist unswitch predicates. Would need to differentiate
+  // start and stop indices. Skip for now as probably not worth for
+  // extra complexity.
```

Comment:
Unswitch tries to do this for the exprs within the unswitch. I was thinking of the pattern:

```
Unswitch:
  Do loads
Inlined:
  Do compute
Unswitch:
  Do stores
```
The first and last unswitch could share much of the computation. However, it may be better to see if we can start moving away from unswitch and just use unroll/vectorize since we now have fewer predicates generally.

Reply:
OK, yes, that could happen. I'll think about it when necessary.